tmail-backend/jmap/extensions/src/main/java/com/linagora/tmail/james/jmap/contact/jCardObjectDeserializer.java

```diff
@@ -0,0 +1,57 @@
+package com.linagora.tmail.james.jmap.contact;
+
+import java.io.IOException;
+import java.util.Iterator;
+
+import com.fasterxml.jackson.core.JacksonException;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import com.google.common.base.Strings;
+
+public class jCardObjectDeserializer extends StdDeserializer<jCardObject> {
+    private static final String FN = "fn";
+    private static final String EMAIL = "email";
+    private static final int PROPERTY_NAME_INDEX = 0;
+    private static final int PROPERTIES_ARRAY_INDEX = 1;
+    private static final int TEXT_PROPERTY_VALUE_INDEX = 3;
+
+    public jCardObjectDeserializer() {
+        this(null);
+    }
+
+    protected jCardObjectDeserializer(Class<?> vc) {
+        super(vc);
+    }
+
+    @Override
+    public jCardObject deserialize(JsonParser p, DeserializationContext ctxt)
+        throws IOException, JacksonException {
+        JsonNode node = p.getCodec().readTree(p);
+
+        JsonNode jCardPropertiesArray = node.get(PROPERTIES_ARRAY_INDEX);
+        Iterator<JsonNode> propertiesIterator = jCardPropertiesArray.iterator();
+
+        String fn = "";
+        String email = "";
+        while (propertiesIterator.hasNext()) {
```

Comment:
Let's work on streams 

(Iterators.toStream(...) )

And let's adopt a 'functional' style!